src/query.ts

```diff
@@ -207,12 +207,19 @@ class Query {
    * const keyQuery = query.filter('__key__', key);
    * ```
    */
-  filter(propertyOrFilter: string | EntityFilter, value?: {} | null): Query;
-  filter(propertyOrFilter: string, operator: Operator, value: {} | null): Query;
-  filter(
-    propertyOrFilter: string | EntityFilter,
+  filter<T extends string>(
+    propertyOrFilter: T | EntityFilter,
+    value?: restrictedValueType<T> | null
+  ): Query;
+  filter<T extends string>(
+    propertyOrFilter: T,
+    operator: Operator,
+    value: restrictedValueType<T> | null
+  ): Query;
+  filter<T extends string>(
+    propertyOrFilter: T | EntityFilter,
     operatorOrValue?: Operator,
```

Comment:
This should be `restrictedValueType<T> | null | Operator`

Reply:
Signature has been updated